app/utils.py

```diff
@@ -382,16 +378,34 @@ def get_app_name():
 
 
 def sanitize_urls(record):
-    """Clean up URLs to protect against XSS."""
+    """Clean up URLs to protect against XSS.
+
+    We check URLs submitted through Person Finder, but bad data might come in
+    through the API.
+    """
+    url_validator = URLValidator()
     # Single-line URLs.
     for field in ['photo_url', 'source_url']:
         url = getattr(record, field, None)
-        if url and not url_is_safe(url):
+        if not url:
+            continue
+        try:
+            url_validator(url)
+        except ValidationError:
             setattr(record, field, None)
     # Multi-line URLs.
     for field in ['profile_urls']:
         urls = (getattr(record, field, None) or '').splitlines()
-        sanitized_urls = [url for url in urls if url and url_is_safe(url)]
+        sanitized_urls = []
+        for url in urls:
+            if not url:
+                continue
```

Comment:
not sure if this is a python thing but in general, I think I would prefer:
```py
for url in urls:
  if url: 
   # validate the url
```

Reply:
It's not a Python thing, changed.